db/migrations/202407151352_autoincrement.go

```diff
@@ -0,0 +1,80 @@
+package migrations
+
+import (
+	_ "embed"
+
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+// This migration (inside a DB transaction),
+// - Adds AUTOINCREMENT to the primary key of:
+// - apps, app_permissions, request_events, response_events, user_configs
+//
+// request_events and response_events are not critical (and also payments are dropped in the same release)
+// so we just drop those tables and re-create them.
+var _202407151352_autoincrement = &gormigrate.Migration{
+	ID: "202407151352_autoincrement",
+	Migrate: func(db *gorm.DB) error {
+
+		if err := db.Transaction(func(tx *gorm.DB) error {
+
+			// drop old request and response event tables
+			if err := tx.Exec(`
+DROP TABLE request_events;
+DROP TABLE response_events;
+`).Error; err != nil {
+				return err
+			}
+
+			// User configs - create new table, copy old values, delete old table, rename new table
+			if err := tx.Exec(`
+CREATE TABLE "user_configs_2" ("id" integer PRIMARY KEY AUTOINCREMENT, "key" text NOT NULL UNIQUE, "value" text, "encrypted" numeric, created_at datetime,updated_at datetime);
+INSERT INTO user_configs_2 (id, key, value, encrypted, created_at, updated_at) SELECT id, key, value, encrypted, created_at, updated_at FROM user_configs;
+DROP TABLE user_configs;
+ALTER TABLE user_configs_2 RENAME TO user_configs;
+`).Error; err != nil {
+				return err
+			}
+
+			// Apps & app permissions (interdependent)
+			// create new tables, copy old values, delete old tables, rename new tables, create new indexes
+			if err := tx.Exec(`
+CREATE TABLE apps_2 (id integer PRIMARY KEY AUTOINCREMENT,name text,description text,nostr_pubkey text UNIQUE,created_at datetime,updated_at datetime, isolated boolean);
+INSERT INTO apps_2 (id, name, description, nostr_pubkey, created_at, updated_at, isolated) SELECT id, name text, description, nostr_pubkey, created_at, updated_at, isolated FROM apps;
```

Comment:
do we already have the `isolated` column? this is because it's not into master, and in the other branch we will have it?

Reply:
@bumi yes, this is branched off (and will be merged into) the dynamic budgets PR.

Also, in that PR, the transactions table correctly has autoincrement on its primary key